tokio/src/sync/mutex.rs

```diff
@@ -130,7 +132,10 @@ pub struct Mutex<T: ?Sized> {
 /// The lock is automatically released whenever the guard is dropped, at which
 /// point `lock` will succeed yet again.
 pub struct MutexGuard<'a, T: ?Sized> {
-    lock: &'a Mutex<T>,
+    s: &'a semaphore::Semaphore,
+    data: *mut T,
+    // Needed to tell the borrow checker that we are holding a `&mut T`
+    marker: marker::PhantomData<&'a mut T>,
```

Comment:
AFAICT, this field is never used, so it should be: 
```suggestion
    _marker: marker::PhantomData<&'a mut T>,
```